src/internal/connector/discovery/api/users.go

```diff
@@ -165,6 +166,78 @@ func (c Users) GetByID(ctx context.Context, identifier string) (models.Userable,
 	return resp, err
 }
 
+// GetUserInfo provide information about a user like- userPurpose/type,
+// if mailbox and onedrive is present for user.
+func (c Users) GetUserInfo(ctx context.Context, userID string) (string, bool, bool, string, error) {
```

Comment:
A couple issues here: 
* stuttering: `users.GetUserInfo` is redundant.  `users.GetInfo` is more sensible.
* collision: `users.GetInfo` already exists.  Having both funcs makes determining their distinct purposes difficult.
* Also, again (trying not to say this too many times, sorry), they could be combined.